src/compiler/core.ts

```diff
@@ -2020,7 +2021,7 @@ namespace ts {
         };
     }
 
-    export function matchFiles(path: string, extensions: string[], excludes: string[], includes: string[], useCaseSensitiveFileNames: boolean, currentDirectory: string, getFileSystemEntries: (path: string) => FileSystemEntries): string[] {
+    export function matchFiles(path: string, extensions: string[], excludes: string[], includes: string[], useCaseSensitiveFileNames: boolean, currentDirectory: string, depth: number | undefined, getFileSystemEntries: (path: string) => FileSystemEntries): string[] {

```

Comment:
What was the reason for intentionally breaking the signature here (the only answer above to the question is that it is supposedly private)? Regardless of the `/* @internal */` comment this method is exported (without an underscore or other indication of it being private) on the main export (e.g. `import { matchFiles } from 'typescript'`). 

This is definitely a breaking change and makes life difficult for tooling that must support both 2.4 and 2.5.  See https://github.com/tildeio/broccoli-typescript-compiler/issues/49.